internal/cli/kraft/net/create/create.go

```diff
@@ -109,3 +104,102 @@ func (opts *CreateOptions) Run(ctx context.Context, args []string) error {
 
 	return nil
 }
+
+func findFreeNetwork(ctx context.Context) (string, error) {
+	type networkPoolEntry struct {
+		Subnet string
+		Size   int
+	}
+
+	defaultNetworkPool := []networkPoolEntry{
+		{"172.17.0.0/16", 16},
+		{"172.18.0.0/16", 16},
+		{"172.19.0.0/16", 16},
+		{"172.20.0.0/14", 16},
+		{"172.24.0.0/14", 16},
+		{"172.28.0.0/14", 16},
+		{"192.168.0.0/16", 20},
+	}
```

Comment:
This should probably be a globally exported variable coming out of `machine/network`

Reply:
Similarly, the `findFreeNetwork` could be a general-purpose method which accepts an input set of networks and an available pool of networks.

Finally, the `net.Interfaces()` should be specific to the underlying driver.

Reply:
Thanks for the review @nderjung! Can you please elaborate a bit on your last point:
> Finally, the net.Interfaces() listing (input networks) should be specific to the underlying driver.

Why is this the case? My idea was that listing all the interfaces on the host, we would be able to identify all subnetworks that are already takes, which we should avoid regardless of the driver we are using for the newly created network.

Reply:
Go's `net.Interfaces()` method is a general-purpose method for returning all interfaces from the host (and has OS-specific implementations) and therefore not necessarily ones under control by the driver.  We can a). shortcut having to query the host entirely by referencing the pool by the driver, making this implementation OS-agnostic and therefore driver-dependent and b). guarantee that a generated "free" IP comes from the driver and therefore guaranteed to work under its supervision.

Ultimately, it's driver which should be responsible for announcing the available pool of networks.  One way tackle this could be to have an IPAM pool which is specified as part of the `NetworkSpec`.  This gives the user the ability to customize the available IPs but also the underlying driver implementation to return the available set.

Check out how CNI is doing something like this for some ideas: https://www.cni.dev/plugins/current/ipam/host-local/

Ultimately, this configuration would be either a). passed in by the user in some form and b). potentially manipulated by an external driver implementation.  For example, the `NetworkService.Get` method may return a modified pool range because it's invoking an external driver program which is outputting this information; the driver is external and therefore its IPAM range is not managed by KraftKit becauase it is configured externally.

Reply:
In fact this probably means that there is an input `NetworkSpec` IPAM address pool and a `NetworkStatus` with current available pool, the latter of which is used to derive a free IP.